src/frontends/paddle/src/internal/pass/transform_fakequantize.cpp

```diff
@@ -0,0 +1,110 @@
+
+#include "default_opset.hpp"
+#include "openvino/pass/pattern/op/label.hpp"
+
+using namespace ov::frontend::paddle::op::default_opset;
+using namespace ov;
+using namespace ov::pass;
+using namespace ov::frontend::paddle::op;
+
+/*
+                                                                       x
+                                                                   |  |  |  |
+                                                                   |  |  |  |
+                                 /                                 V  v  v  v
+                                |              +-----------+     +-----------+
+                                |   scale -->  | Multiply  | --> |  Divide   |
+    +------------------+        |              +-----------+     +-----------+
+    |                  |        |                                      |
+    |  quantize_linear |  -->   |                                      v
+    |                  |        |                                +-----------+
+    +------------------+        |                                |   Round   |
+                                |                                +-----------+                  |   |   |   |
+                                |                                      |                        v   v   v   v
+                                |                                      v                    +-------------------+
+                                |                                +-----------+              |                   |
+                                |                                |   Clamp   |     === >    |   FakeQuantize    |
+                                 \                               +-----------+              |                   |
+                                                                       |                    +-------------------+
+                                 /                                     v                        |   |   |   |
+     +------------------+       |                                +-----------+                  v   v   v   v
+     |                  |       |                                |  Convert  |
+     |dequantize_linear |  -->  |                                +-----------+
+     |                  |       |                                      |
+     +------------------+       |                                      v
+                                |              +-----------+     +-----------+
+                                |   scale -->  | Multiply  | --> | Multiply  |
+                                |              +-----------+     +-----------+
+                                |                                 |  |  |  |
+                                 \                                |  |  |  |
+                                                                  v  v  v  v
+                                                                       Y
+*/
```

Comment:
> As I can see from the code, the rounding mode is never used (and cannot really be used if we want to converge to FQ). That means you can mathematically represent quantize_linear part differently -- without round involvement -- in the same approach as ONNX FE uses. Please check if decomposition used in ONNX here: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/quantize_linear.cpp is equivalent to what you are trying to achieve (with rounding mode removed).
> 
> And the same is true (but for a reason that doesn't involve this rounding mode) for Dequantize part, where you can reuse this thing: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/dequantize_linear.cpp.
> 
> And if you are using such decompositions, I believe you won't longer need this transformation to fuse everything to FQ because we already have it here: https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/convert_quantize_dequantize.cpp. This transformation will be called as a part of MOC or inference plugin transformation pipeline (should be checked).
> 
> This is what I meant in my first e-mail when referring to similarities with ONNX. If it is incorrect, please be more specific why it cannot be implemented in that way.

- For round mode, do not worry about it for now because the PaddleSlim will set it to a default value and have no plan to use other value.
- For removing the custom pass in PDPD FE, I agree with you that we should reuse the transformation codes and shouldn't reinvent it as you said. So I create the new branch https://github.com/xczhai/openvino/tree/paddle_slim_update. It can be replaced by the https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/convert_quantize_dequantize.cpp. [convert_quantize_dequantize.cpp] but the generated ir runs in FP32 kernel instead of INT8. Please help take a look.